cmd/notify-mailer/main.go

```diff
@@ -421,56 +487,66 @@ func main() {
 	flag.Parse()
 	if *from == "" || *subject == "" || *bodyFile == "" || *configFile == "" ||
-		*recipientListFile == "" {
+		(*recipientListCSV == "" && *recipientListTSV == "") || (*recipientListCSV != "" && *recipientListTSV != "") {
 		flag.Usage()
 		os.Exit(1)
 	}
 
 	configData, err := ioutil.ReadFile(*configFile)
-	cmd.FailOnError(err, fmt.Sprintf("Reading %q", *configFile))
+	cmd.FailOnError(err, "Couldn't read JSON config file")
+
+	// Parse JSON config.
+	type config struct {
+		NotifyMailer struct {
+			DB cmd.DBConfig
+			cmd.SMTPConfig
+		}
+		Syslog cmd.SyslogConfig
+	}
+
 	var cfg config
 	err = json.Unmarshal(configData, &cfg)
-	cmd.FailOnError(err, "Unmarshaling config")
-	err = features.Set(cfg.NotifyMailer.Features)
-	cmd.FailOnError(err, "Failed to set feature flags")
+	cmd.FailOnError(err, "Couldn't unmarshal JSON config file")
 
 	log := cmd.NewLogger(cfg.Syslog)
 	defer log.AuditPanic()
 
-	dbURL, err := cfg.NotifyMailer.DB.URL()
-	cmd.FailOnError(err, "Couldn't load DB URL")
+	// Setup database client.
 	dbSettings := sa.DbSettings{
-		MaxOpenConns: 10,
+		MaxOpenConns:    cfg.NotifyMailer.DB.MaxOpenConns,
+		MaxIdleConns:    cfg.NotifyMailer.DB.MaxIdleConns,
+		ConnMaxLifetime: cfg.NotifyMailer.DB.ConnMaxLifetime.Duration,
+		ConnMaxIdleTime: cfg.NotifyMailer.DB.ConnMaxIdleTime.Duration,
 	}
+
+	dbURL, err := cfg.NotifyMailer.DB.URL()
+	cmd.FailOnError(err, "Couldn't load DB URL")
+
 	dbMap, err := sa.NewDbMap(dbURL, dbSettings)
-	cmd.FailOnError(err, "Could not connect to database")
+	cmd.FailOnError(err, "Couldn't create database connection")
 
-	// Load email body
-	body, err := ioutil.ReadFile(*bodyFile)
-	cmd.FailOnError(err, fmt.Sprintf("Reading %q", *bodyFile))
-	template, err := template.New("email").Parse(string(body))
-	cmd.FailOnError(err, fmt.Sprintf("Parsing template %q", *bodyFile))
+	// Load and parse message details.
+	template, err := template.New("email").ParseFiles(*bodyFile)
+	cmd.FailOnError(err, "Couldn't parse message template")
 
 	address, err := mail.ParseAddress(*from)
 	cmd.FailOnError(err, fmt.Sprintf("Parsing %q", *from))
 
-	recipients, err := readRecipientsList(*recipientListFile)
-	cmd.FailOnError(err, fmt.Sprintf("Reading %q", *recipientListFile))
-
-	targetRange := interval{
-		start: *start,
-		end:   *end,
-	}
+	recipients, err := makeRecipientList(recipientListCSV, recipientListTSV)
+	cmd.FailOnError(err, "Couldn't populate recipients from provided recipient list")
 
 	var mailClient bmail.Mailer
 	if *dryRun {
-		log.Infof("Doing a dry run.")
+		log.Infof("Starting notify-mailer in dry-run mode")
 		mailClient = bmail.NewDryRun(*address, log)
 	} else {
-		smtpPassword, err := cfg.NotifyMailer.PasswordConfig.Pass()
-		cmd.FailOnError(err, "Failed to load SMTP password")
+		smtpPassword, err := cfg.NotifyMailer.SMTPConfig.PasswordConfig.Pass()
+		cmd.FailOnError(err, "Couldn't load SMTP password from file")
+
+		log.Infof("Starting notify-mailer")
```

Comment:
We normally do this like:

```suggestion
		log.Infof(cmd.VersionString())
```